tests/e2e_tests/noisemap_cal_e2e.py

```diff
@@ -41,68 +50,138 @@ def test_noisemap_calibration_from_l1(tvacdata_path, e2eoutput_path):
+    for i, dset in enumerate(splitl1):
+        stackl1_arr.append(dset.all_data[:10]) #get first 10 frames, to speed up runs
+        for j in range(len(dset.all_data[:10])):
+            stack_arr_files.append(dset.frames[j].filepath)
+        exptime_arr.append(splitl1_params[i][0])
+        gain_arr.append(splitl1_params[i][1])
+    stackl1_arr = np.stack(stackl1_arr)
+    kgain_arr = [8.7]*len(exptime_arr)
+    
+    exptime_arr = np.array(exptime_arr)
+    gain_arr = np.array(gain_arr)
+    kgain_arr = np.array(kgain_arr)
     
+    output_filenamel1 = os.path.split(stack_arr_files[0])[1][:-5] + '_DetectorNoiseMaps.fits'
+    #updates to L2a:
+    output_filename = output_filenamel1.replace('L1','L2a',1)
+
+    (F_map, C_map, D_map, bias_offset, F_image_map, C_image_map,
+                D_image_map, Fvar, Cvar, Dvar, read_noise, R_map, F_image_mean,
+                C_image_mean, D_image_mean, unreliable_pix_map) = \
+    calibrate_darks_lsq(stackl1_arr, gain_arr, exptime_arr, kgain_arr, fwc_em_e, fwc_pp_e,
+                meta_path, nonlin_path, Nem = 604, telem_rows=telem_rows, 
+                sat_thresh=0.7, plat_thresh=0.7, cosm_filter=1, cosm_box=3,
+                cosm_tail=10, desmear_flags=None, rowreadtime=223.5e-6)
+    ##########
+
     ###### Setup necessary calibration files
     # Create necessary calibration files
     # we are going to make calibration files using
     # a combination of the II&T nonlinearty file and the mock headers from
     # our unit test version
-    pri_hdr, ext_hdr = mocks.create_default_headers()
-    ext_hdr["DRPCTIME"] = time.Time.now().isot
-    ext_hdr['DRPVERSN'] =  corgidrp.__version__
     mock_input_dataset = data.Dataset(mock_cal_filelist)
 
     this_caldb = caldb.CalDB() # connection to cal DB
 
+    pri_hdr, ext_hdr = mocks.create_default_headers()
+    ext_hdr["DRPCTIME"] = time.Time.now().isot
+    ext_hdr['DRPVERSN'] =  corgidrp.__version__
+
     # Nonlinearity calibration
     nonlin_dat = np.genfromtxt(nonlin_path, delimiter=",")
     nonlinear_cal = data.NonLinearityCalibration(nonlin_dat, pri_hdr=pri_hdr, ext_hdr=ext_hdr,
                                                 input_dataset=mock_input_dataset)
     nonlinear_cal.save(filedir=noisemap_outputdir, filename="mock_nonlinearcal.fits" )
     this_caldb.create_entry(nonlinear_cal)
 
+
     # KGain
-    kgain_val = 8.7 
+    kgain_val = 8.7 # From TVAC-20 noise characterization measurements
     kgain = data.KGain(np.array([[kgain_val]]), pri_hdr=pri_hdr, ext_hdr=ext_hdr, 
                     input_dataset=mock_input_dataset)
     kgain.save(filedir=noisemap_outputdir, filename="mock_kgain.fits")
     this_caldb.create_entry(kgain)
 
 
     ####### Run the walker on some test_data
-
     template = "l1_to_l2a_noisemap.json"
-    walker.walk_corgidrp(l1_data_filelist, "", noisemap_outputdir,template=template)
+    walker.walk_corgidrp(stack_arr_files, "", noisemap_outputdir,template=template)
```

Comment:
same as for the other e2e tests. It would be good to keep intermediate variables separate between the two. This is to improve test clarity.